stok.js

```diff
@@ -0,0 +1,78 @@
+'use strict';
+
+const Hapi = require('hapi');
+
+class Stok {
+  constructor(options) {
+    this.options = options;
+  }
+
+  createServer(routes) {
+    return new Promise((resolve, reject) => {
+      const server = new Hapi.Server();
+      server.connection({ port: this.options.web.port });
+
+      // Monkey patch the route method and add error handling to
+      // all of our routes
+      server.route = ((original) => {
+        routes => {
+          routes.forEach(route => {
+            let handler = route.handler;
+            route.handler = ((request, reply) => {
+              handler(request, reply)
+                .catch(error => {
+                  reply(error);
+                });
+            });
+          });
+          original.call(server, routes);
+        };
+      })(server.route(routes));
+
+      server.register(getLoggingSettings(this.options.web.logging.filePath), error => {
+        if (error) {
+          reject(error);
+        }
+      });
+
+      resolve(server);
```

Comment:
This is problematic because the logger registration is async and can fail after you've resolved here. But my next comment will propose a fix anyway :-)